paasta_tools/setup_kubernetes_job.py

```diff
@@ -0,0 +1,179 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Usage: ./setup_kubernetes_job.py <service.instance> [options]
+
+Command line options:
+
+- -d <SOA_DIR>, --soa-dir <SOA_DIR>: Specify a SOA config dir to read from
+- -v, --verbose: Verbose output
+"""
+import argparse
+import logging
+import sys
+from typing import List
+from typing import Optional
+from typing import Tuple
+
+from paasta_tools.kubernetes_tools import create_deployment
+from paasta_tools.kubernetes_tools import ensure_paasta_namespace
+from paasta_tools.kubernetes_tools import KubeClient
+from paasta_tools.kubernetes_tools import KubeDeployment
+from paasta_tools.kubernetes_tools import list_all_deployments
+from paasta_tools.kubernetes_tools import load_kubernetes_service_config_no_cache
+from paasta_tools.kubernetes_tools import update_deployment
+from paasta_tools.utils import decompose_job_id
+from paasta_tools.utils import DEFAULT_SOA_DIR
+from paasta_tools.utils import InvalidJobNameError
+from paasta_tools.utils import load_system_paasta_config
+from paasta_tools.utils import NoConfigurationForServiceError
+from paasta_tools.utils import NoDeploymentsAvailable
+from paasta_tools.utils import NoDockerImageError
+from paasta_tools.utils import SPACER
+
+log = logging.getLogger(__name__)
+
+
+def parse_args() -> argparse.Namespace:
+    parser = argparse.ArgumentParser(description='Creates marathon jobs.')
+    parser.add_argument(
+        'service_instance_list', nargs='+',
+        help="The list of marathon service instances to create or update",
+        metavar="SERVICE%sINSTANCE" % SPACER,
+    )
+    parser.add_argument(
+        '-d', '--soa-dir', dest="soa_dir", metavar="SOA_DIR",
+        default=DEFAULT_SOA_DIR,
+        help="define a different soa config directory",
+    )
+    parser.add_argument(
+        '-v', '--verbose', action='store_true',
+        dest="verbose", default=False,
+    )
+    args = parser.parse_args()
+    return args
+
+
+def main() -> None:
+    args = parse_args()
+    soa_dir = args.soa_dir
+    if args.verbose:
+        logging.basicConfig(level=logging.DEBUG)
+    else:
+        logging.basicConfig(level=logging.WARNING)
+
+    # system_paasta_config = load_system_paasta_config()
+    kube_client = KubeClient()
+
+    ensure_paasta_namespace(kube_client)
+    num_failed_deployments = setup_kube_deployments(
+        kube_client=kube_client,
+        service_instances=args.service_instance_list,
+        soa_dir=soa_dir,
+    )
+    sys.exit(1 if num_failed_deployments else 0)
+
+
+def setup_kube_deployments(
+    kube_client: KubeClient,
+    service_instances: List[str],
+    soa_dir: str=DEFAULT_SOA_DIR,
+) -> int:
```

Comment:
if we're not doing anything with the number here, then can we return a boolean instead?